pkg/pool-utils/test/foundry/BasePoolFactory.t.sol

```diff
@@ -103,6 +103,27 @@ contract BasePoolFactoryTest is BaseVaultTest {
         testFactory.manualRegisterPoolWithFactory(newPool);
 
         assertTrue(testFactory.isPoolFromFactory(newPool), "Pool is not registered with factory");
+        assertEq(testFactory.getPools(0, 1)[0], newPool, "Pools list does not contain the new pool");
+    }
+
+    function testRegisterMultiplePools() public {
```

Comment:
Can we add tests to get partial lists, and to check the edges? (length > count, end > count, etc)